### DISCUSSION TITLE
Timing of adopting new language features

### DISCUSSION THREAD

**Discussion by chris-martin** [haskellfoundation/HaskellSchool#53] [General]
We need to strike some balance between:

- Not using features that the reader's installed compiler doesn't have
- Not becoming outdated and cluttered with things that are no longer necessary

In particular the question on my mind is when to stop telling users to enable GHC2021 extensions and instead just to set `-XGHC2021` (available in GHC 9.2). I think it's probably too soon at the moment, but it would be good if we could decide in advance when the time will be right.

A general policy like "start assuming features are available if the compiler was released at least X months ago" might be the way to go here? As an initial proposal, I'll suggest 18 months, because that's about how long it took me from the release of GHC 9.0 to the time when there was enough library support to actually use it.

**Comment by Kleidukos** [comment]
We have a fortunately small surface to cover when it comes to updated dependencies, at Haskell School. I think as well that a delay of 18 months is reasonable.
